src/main/java/spoon/reflect/path/impl/CtPathImpl.java

```diff
@@ -33,9 +33,70 @@ public <T extends CtElement> List<T> evaluateOn(CtElement... startNode) {
 		for (CtPathElement element : elements) {
 			filtered = element.getElements(filtered);
 		}
+		if (filtered.isEmpty()) {
+			List<String> cls_name_list = new LinkedList<>();
+			CtType<?> ctType = null;
+			for (CtPathElement element : elements) {
+				if (element instanceof CtRolePathElement) {    // search by CtRolePathElement
+					Collection<String> values = ((CtRolePathElement) element).getArguments().values();
+					String val = null;
+					if (values.iterator().hasNext()) val = values.iterator().next();
+					if (val != null) {
+						if (CtRole.SUB_PACKAGE.equals(((CtRolePathElement) element).getRole())
+								|| CtRole.CONTAINED_TYPE.equals(((CtRolePathElement) element).getRole()))
+							cls_name_list.add(val);
+
+						Class<?> cls = getJdkClass(String.join(".", cls_name_list));
+						if (cls != null) {
+							if (ctType == null) ctType = new TypeFactory().get(cls);
+							else {
+								CtElement result = null;
+								if (CtRole.METHOD.equals(((CtRolePathElement) element).getRole())) {
+									result = ctType.getMethodBySignature(val);
+								}
+								if (CtRole.CONSTRUCTOR.equals(((CtRolePathElement) element).getRole())) {
+									result = ((CtClass) ctType).getConstructorBySignature(val);
+								}
+								if (CtRole.FIELD.equals(((CtRolePathElement) element).getRole())) {
+									result = ctType.getField(val);
+								}
+								if (result != null) filtered.add(result);
+							}
+						}
+					}
+				}
+			}
+			if (filtered.isEmpty() && ctType != null) filtered.add(ctType);
+		}
 		return (List<T>) filtered;
 	}
 
+	private Class<?> getJdkClass(String name) {
```

Comment:
Note: This is duplicated in a few places in spoon

Reply:
Yes, I found similar one: `TypeFactory.get(final String qualifiedName)`.
However, this method doesn't support shadow elements while another reload method `TypeFactory.get(Class<?> cl)` does.

So the question is that we need a function that transform `String` to `Class<?>`, as the method above did,
or add extra logic in `TypeFactory.get(final String qualifiedName)` to process shadow elements intendly.